services/httpd/handler.go

```diff
@@ -517,25 +518,36 @@ func (h *Handler) serveWrite(w http.ResponseWriter, r *http.Request, user *meta.
 	}
 
 	h.statMap.Add(statPointsWrittenOK, int64(len(points)))
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
+
+	// Clients requesting gzipped response will receive StatusOK and
+	// minimal gzip response.
 }
 
 // serveOptions returns an empty response to comply with OPTIONS pre-flight requests
 func (h *Handler) serveOptions(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
 }
 
 // servePing returns a simple response to let the client know the server is running.
 func (h *Handler) servePing(w http.ResponseWriter, r *http.Request) {
 	h.statMap.Add(statPingRequest, 1)
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
 }
 
 // serveStatus has been depricated
 func (h *Handler) serveStatus(w http.ResponseWriter, r *http.Request) {
 	h.Logger.Printf("WARNING: /status has been depricated.  Use /ping instead.")
 	h.statMap.Add(statStatusRequest, 1)
-	w.WriteHeader(http.StatusNoContent)
+	if r.Header.Get("Accept-Encoding") != "gzip" {
+		w.WriteHeader(http.StatusNoContent)
+	}
```

Comment:
@mark-rushakoff it was my understanding from your original issue that clients are requesting gzipped responses and then processing the responses as gzipped regardless of `Content-Encoding` header. 

If the clients are checking the `Content-Type` header then yes we could never set it for the `write` endpoint, which would be a simpler solution. 👍